tests/unit/s2n_self_talk_ktls_test.c

```diff
@@ -87,11 +87,14 @@ static S2N_RESULT s2n_new_inet_socket_pair(struct s2n_test_io_pair *io_pair)
     RESULT_ENSURE_GTE(pid, 0);
     if (pid == 0) {
         RESULT_ENSURE_EQ(connect(io_pair->client, (struct sockaddr *) &saddr, addrlen), 0);
+        EXPECT_SUCCESS(s2n_io_pair_close(io_pair));
         ZERO_TO_DISABLE_DEFER_CLEANUP(io_pair);
```

Comment:
Why would you zero to avoid cleanup (which I assume is s2n_io_pair_close) after you've already closed? That suggests the test didn't want the io_pair closed here.

Reply:
`ZERO_TO_DISABLE_DEFER_CLEANUP(io_pair);` should be removed and `io_pair` should be closed.

This is the end of the child process, so proper cleanup should be done. It's worth to note that `DEFER_CLEANUP` for `io_pair` wouldn't work, since the process didn't exit in the same scope as where io_pair was created.

Reply:
But these tests assume that the client and server sockets will still be valid in the parent process, right? Is it okay to call `close` on the file descriptors for these sockets in the child process and continue using the sockets in the parent process?

Reply:
I have proposed to remove the fork and do connect and accept in a single process. I have added my explanation for such change in the PR comments.